packages/adminpanel-v2/src/components/customerpage/customerdetails/headers/customerinfo/CustomerInfo.svelte

```diff
@@ -23,17 +26,26 @@
 	};
 
 	let addFunction = () => {
-		if (add && customer === undefined) {
+		if (add && !(newCustomerName && newCustomerNumber)) {
 			throw new Error('Not valid state for program...');
 		}
-		if (add && customer) {
-			customers.updateCustomer(customer);
-		}
-		add = !add;
+		else if (add) {
+			customers.addNewCustomer(newCustomerName!, newCustomerNumber!)
+			add = !add;
```

Comment:
Consider using an enum instead of a boolean.

Something like:

```
enum State {
   EditState,
   ViewState
}

let currentState = State.ViewState;
```